File: modules/core/src/main/java/de/deutschebahn/bahnhoflive/ui/station/StationNavigation.java
/*
 * SPDX-FileCopyrightText: 2020 DB Station&Service AG <bahnhoflive-opensource@deutschebahn.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package de.deutschebahn.bahnhoflive.ui.station;

import de.deutschebahn.bahnhoflive.ui.station.timetable.TimetablesFragment;

public interface StationNavigation extends TimetablesFragment.Host {
    @Override
    void showTimetablesFragment(boolean localTransport, boolean arrivals, String trackFilter);

    void showShopsFragment();

    void showFeedbackFragment();

    void showSettingsFragment();

    void showContentSearch();

    void showLocalTransport();

    void showLocalTransportTimetableFragment();

    void showStationFeatures();

    void showNewsDetails(final int newsIndex);

    void showOccupancyExplanation();

    void showInfoFragment(boolean clearStack);

    void showElevators();

    void showParkings();

    void showAccessibility();

    void showRailReplacement();

    void showLockers();
}

File: modules/core/src/main/java/de/deutschebahn/bahnhoflive/ui/station/StationNavigation.kt
/*
 * SPDX-FileCopyrightText: 2020 DB Station&Service AG <bahnhoflive-opensource@deutschebahn.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
package de.deutschebahn.bahnhoflive.ui.station

import de.deutschebahn.bahnhoflive.ui.station.timetable.TimetablesFragment

interface StationNavigation : TimetablesFragment.Host {
    override fun showTimetablesFragment(
        localTransport: Boolean,
        arrivals: Boolean,
        trackFilter: String?
    )

    fun showShopsFragment()
    fun showFeedbackFragment()
    fun showSettingsFragment()
    fun showContentSearch()
    fun showLocalTransport()
    fun showLocalTransportTimetableFragment()
    fun showStationFeatures()
    fun showNewsDetails(newsIndex: Int)
    fun showOccupancyExplanation()
    fun showInfoFragment(clearStack: Boolean)
    fun showElevators()
    fun showParkings()
    fun showAccessibility()
    fun showRailReplacement()
    fun showRailReplacementStopPlaceInformation()
    fun showRailReplacementDbCompanionInformation()
    fun showLockers(removeFeaturesFragment: Boolean)
    fun showMobilityServiceNumbers()
    fun showInfo(serviceContentType: String, removeFeaturesFragment: Boolean)
//    fun showBhfLiveNext()
    fun showDbCompanionHelp()
}
